src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt

```diff
@@ -219,10 +219,11 @@ class UserService(
       is ClientResult.Failure -> staffUserDetailsResponse.throwException()
     }
 
-    if (deliusUser.teams?.firstOrNull()?.borough?.code != null && user.probationDeliveryUnit?.deliusCode !== deliusUser.teams.firstOrNull()?.borough?.code) {
-      when (val probationDeliveryUnit = findProbationDeliveryUnit(deliusUser.teams)) {
+    val userTeams = orderUserDeliusTeams(deliusUser)
+    if (userTeams?.firstOrNull()?.borough?.code != null && user.probationDeliveryUnit?.deliusCode !== userTeams.firstOrNull()?.borough?.code) {
+      when (val probationDeliveryUnit = findProbationDeliveryUnit(userTeams)) {
         null -> {
-          throw Exception("Unable to find community API borough code ${deliusUser.teams.first().borough.code} in CAS")
+          throw Exception("Unable to find community API borough code ${userTeams.first().borough.code} in CAS")
```

Comment:
userTeams.first().borough.code or variations on it are used in 3 places in this function. Maybe line 221 could be changed to :

`val firstBoroughCode = orderUserDeliusTeams(deliusUser)?.firstOrNull()?.borough?.code`

And then firstBoroughCode could be used through the if block?

Reply:
I need userTeams as I use it in line 224

Reply:
my suggestion was to capture in a variable for first usage on onwards but this is just minor duplication so happy to merge if you're happy

Reply:
I changed the code by removing the function findProbationDeliveryUnit and get the last Borough code for the user as you suggested